avm/src/daemon/daemon.rs

```diff
@@ -26,29 +30,40 @@ use crate::vm::run::run;
 pub static CLUSTER_SECRET: OnceCell<Option<String>> = OnceCell::new();
 
 #[cfg(target_os = "linux")]
-async fn get_private_ip() -> String {
+async fn get_private_ip() -> Result<String, String> {
   let res = Command::new("hostname").arg("-I").output().await;
   let err = "Failed to execute `hostname`";
-  let stdout = res.expect(err).stdout;
-  String::from_utf8(stdout)
-    .expect(err)
-    .trim()
-    .split_whitespace()
-    .next()
-    .unwrap()
-    .to_string()
+  error!(116, "TEST priv ip").await;
+  match res {
+    Ok(res) => {
+      let stdout = res.stdout;
+      let private_ip = String::from_utf8(stdout);
+      match private_ip {
+        Ok(private_ip) => match private_ip.trim().split_whitespace().next() {
+          Some(private_ip) => Ok(private_ip.to_string()),
+          None => return Err("No ip found".to_string()),
+        },
+        Err(err) => return Err(err.to_string()),
+      }
+    }
+    Err(_) => return Err(err.to_string()),
+  }
 }
 
 #[cfg(not(target_os = "linux"))]
-async fn get_private_ip() -> String {
+async fn get_private_ip() -> Result<String, String> {
   panic!("`hostname` command does not exist in this OS");
 }
 
 async fn post_v1(endpoint: &str, body: Value) -> String {
   let resp = deploy::post_v1(endpoint, body).await;
   match resp {
     Ok(res) => res,
-    Err(err) => format!("{:?}", err),
+    Err(err) => {
+      let err = format!("{:?}", err);
+      error!(121, "{:?}", err).await;
```

Comment:
Similar comment as what @cdmistman made in the other PR. These should be in a named enum so the error codes make sense to the reader. No perf impact to doing so.